Milyli.ScriptRunner.Core/Services/RelativityScriptRunner.cs

```diff
@@ -1,43 +1,101 @@
+		private readonly IRelativityScriptRepository relativityScriptRepository;
+		private readonly IRelativityScriptProcessor relativityScriptProcessor;
+		private readonly ISearchTableManager searchTableManager;
+		private readonly IHelper helper;
+
+		public RelativityScriptRunner(
+			IJobScheduleRepository jobScheduleRepository,
+			IRelativityClientFactory relativityClient,
+			IRelativityScriptRepository relativityScriptRepository,
+			IRelativityScriptProcessor relativityScriptProcessor,
+			ISearchTableManager searchTableManager,
+			IHelper helper)
 		{
 			this.jobScheduleRepository = jobScheduleRepository;
 			this.relativityClient = relativityClient;
 			this.relativityScriptRepository = relativityScriptRepository;
+			this.relativityScriptProcessor = relativityScriptProcessor;
+			this.searchTableManager = searchTableManager;
+			this.helper = helper;
 		}
 
-		private static NLog.Logger Logger
-		{
-			get
-			{
-				return NLog.LogManager.GetLogger("Default");
-			}
-		}
-
+		private static NLog.Logger Logger => NLog.LogManager.GetLogger("Default");
+
+		/// <inheritdoc />
 		public void ExecuteAllJobs(DateTime executionTime)
 		{
 			var schedules = this.jobScheduleRepository.GetJobSchedules(executionTime);
 			Logger.Trace($"found {schedules.Count} jobs to execute");
-			schedules.ForEach(this.ExecuteScriptJob);
-		}
-
+			foreach (var schedule in schedules)
+			{
+				if (schedule.DirectSql)
+				{
+					this.ExecuteDirectSqlJob(schedule);
+				}
+				else
+				{
+					this.ExecuteScriptJob(schedule);
+				}
+			}
+		}
+
+		/// <inheritdoc />
+		public void ExecuteDirectSqlJob(JobSchedule job)
+		{
+			var workspace = new RelativityWorkspace { WorkspaceId = job.WorkspaceId };
+			var searchIds = new List<int>();
+			var activationStatus = this.jobScheduleRepository.StartJob(job);
+
+			if (activationStatus == JobActivationStatus.Started)
+			{
+				try
+				{
+					var inputs = this.jobScheduleRepository.GetJobInputs(job);
+					var script = RelativityHelper.InWorkspace((client, _) => client.Repositories.RelativityScript.ReadSingle(job.RelativityScriptId), workspace, this.relativityClient.GetRelativityClient());
+					var originalInputs = RelativityHelper.InWorkspace((client, _) => client.Repositories.RelativityScript.GetRelativityScriptInputs(script), workspace, this.relativityClient.GetRelativityClient());
+					var scriptSql = script.Body.AsXmlDocument.GetElementsByTagName("action").Item(0).InnerText;
+					scriptSql = this.relativityScriptProcessor.SubstituteGlobalVariables(workspace.WorkspaceId, scriptSql);
+					scriptSql = this.relativityScriptProcessor.SubstituteScriptInputs(inputs, originalInputs, scriptSql, SearchTablePrepend, job.Id);
+					searchIds = this.relativityScriptProcessor.GetSavedSearchIds(inputs, originalInputs).ToList();
+
+					this.searchTableManager.CreateTablesAsync(SearchTablePrepend, workspace.WorkspaceId, searchIds, job.Id, job.MaximumRuntime).GetAwaiter().GetResult();

```

Comment:
I'd like to see this returning the list of materialized search names and their artifact ids, then an additional step for replacing the tokens in the `scriptSql`.